compiler/suggest.nim

```diff
@@ -105,7 +105,7 @@ proc getTokenLenFromSource(conf: ConfigRef; ident: string; info: TLineInfo): int
       result = 0
   elif ident[0] in linter.Letters and ident[^1] != '=':
     result = identLen(line, column)
-    if cmpIgnoreStyle(line[column..column + result - 1], ident) != 0:
+    if cmpIgnoreStyle(line[column..column + result - 1], ident[0..min(result-1,len(ident)-1)]) != 0:
```

Comment:
Why?

Reply:
Because ident may have extra stuff appended when part of a template. E.g. when a variable is part of a template, "`gensymXXX" is appended, where XXX are some numbers. In the previous version of the code, this would return a token length of 0. In turn, this causes the inlay hint to be put in the wrong place. Instead of:
```
template benchmark(benchmarkName: string, code: untyped) =
  block:
    debug "Started...", benchmark = benchmarkName
    let t0: float = epochTime()
    code
    let elapsed: float = epochTime() - t0
    let elapsedStr: string = elapsed.formatFloat(format = ffDecimal, precision = 3)
    debug "CPU Time", benchmark = benchmarkName, time = elapsedStr
```
You would get:
```
template benchmark(benchmarkName: string, code: untyped) =
  block:
    debug "Started...", benchmark = benchmarkName
    let : floatt0 = epochTime()
    code
    let : floatelapsed = epochTime() - t0
    let : stringelapsedStr = elapsed.formatFloat(format = ffDecimal, precision = 3)
    debug "CPU Time", benchmark = benchmarkName, time = elapsedStr
```

Reply:
Sucks, but ok.